source/install/config-ssl-http2-nginx.rst

```diff
@@ -14,95 +14,140 @@ You can use any certificate that you want, but these instructions show you how t
 .. note::
    If Let’s Encrypt is enabled, forward port 80 through a firewall, with `Forward80To443 <https://docs.mattermost.com/administration/config-settings.html#forward-port-80-to-443>`__ ``config.json`` setting set to ``true`` to complete the Let’s Encrypt certification.
 
-**To configure NGINX as a proxy with SSL and HTTP/2:**
+**To configure NGINX as a proxy with SSL and HTTP/2**
+
+If you're looking for additional Let's Encrypt/Certbot assistance you can access their documentation `here <https://certbot.eff.org>`_ .
 
 1. Log in to the server that hosts NGINX and open a terminal window.
-2. Install git.
 
-  If you are using Ubuntu or Debian:
+2. Open the your Mattermost ``nginx.conf`` file as *root* in a text editor and update the IP address in the ``upstream backend`` to point towards Mattermost, and the ``server_name`` to be your domain for Mattermost.
 
-  ``sudo apt-get install git``
+.. note::
+   On Ubuntu this file is located at ``/etc/nginx/sites-available/``. If you don't have this file run ``sudo touch /etc/nginx/sites-available/mattermost``.
+   On CentOS/RHEL this file is located at ``/etc/nginx/conf.d/``. If you don't have this file run ``sudo touch /etc/nginx/conf.d/mattermost``.
+   
+.. code-block:: none
 
-  If you are using RHEL:
+   upstream backend {
+       server {ip}:8065;
+       keepalive 32;
+       }
 
-  ``sudo yum install git``
+   proxy_cache_path /var/cache/nginx levels=1:2 keys_zone=mattermost_cache:10m max_size=3g inactive=120m use_temp_path=off;
 
-3. Clone the Let's Encrypt repository on GitHub.
+   server {
+       listen 80 default_server;
+       server_name mattermost.example.com;
 
-  ``git clone https://github.com/letsencrypt/letsencrypt``
+       location ~ /api/v[0-9]+/(users/)?websocket$ {
+           proxy_set_header Upgrade $http_upgrade;
+           proxy_set_header Connection "upgrade";
+           client_max_body_size 50M;
+           proxy_set_header Host $http_host;
+           proxy_set_header X-Real-IP $remote_addr;
+           proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+           proxy_set_header X-Forwarded-Proto $scheme;
+           proxy_set_header X-Frame-Options SAMEORIGIN;
+           proxy_buffers 256 16k;
+           proxy_buffer_size 16k;
+           client_body_timeout 60;
+           send_timeout 300;
+           lingering_timeout 5;
+           proxy_connect_timeout 90;
+           proxy_send_timeout 300;
+           proxy_read_timeout 90s;
+           proxy_pass http://backend;
+       }
 
-4. Change to the ``letsencrypt`` directory.
+       location / {
+           client_max_body_size 50M;
+           proxy_set_header Connection "";
+           proxy_set_header Host $http_host;
+           proxy_set_header X-Real-IP $remote_addr;
+           proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
```

Comment:
Unless you are nesting proxies, you may want `$remote_addr` for it as well:

```
$proxy_add_x_forwarded_for
the “X-Forwarded-For” client request header field with the $remote_addr variable appended to it, separated by a comma. If the “X-Forwarded-For” field is not present in the client request header, the $proxy_add_x_forwarded_for variable is equal to the $remote_addr variable.
```

Reply:
This is actually the default nginx.conf config we have listed now. Are you saying to modify it to :

```
proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for, $remote_addr;
```

I'm not as familiar with this. Just copy/pasted what we had previously.

Reply:
Let's revisit this later on since it's an unrelated change and will need some additional documentation. We can create a new issue to track this.